lib/console.js

```diff
@@ -35,26 +35,25 @@ function Console(stdout, stderr) {
   }
 }
 
-Console.prototype.log = function() {
-  this._stdout.write(util.format.apply(null, arguments) + '\n');
+Console.prototype.log = function(...args) {
+  this._stdout.write(`${util.format(...args)}\n`);
```

Comment:
Yep, as indicated, much of this change is speculative. The code cleanups
make a few assumptions about what v8 v5 *should* be doing as far as
optimizations but those assumptions have yet to be tested. Those specific
changes can be pulled back out of this if necessary tho.
On Apr 13, 2016 1:42 AM, "Benjamin Gruenbaum" <notifications@github.com>
wrote:

In lib/console.js
<https://github.com/nodejs/node/pull/6174#discussion_r59512094>:

> @@ -35,26 +35,25 @@ function Console(stdout, stderr) {
>    }
>  }
>
> -Console.prototype.log = function() {
> -  this._stdout.write(util.format.apply(null, arguments) + '\n');
> +Console.prototype.log = function(...args) {
> +  this._stdout.write(`${util.format(...args)}\n`);

Isn't ...args still slightly slower than .apply?

Not that it would probably be measurable given log.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub
<https://github.com/nodejs/node/pull/6174/files/2ddd0562b0534519eac99f0917ebec9cb3b1fe7a#r59512094>